src/controllers/device.controller.ts

```diff
@@ -0,0 +1,27 @@
+import type {
+  Handler,
+  Request,
+  Response,
+} from 'express';
+import { deviceService } from '../services';
+import { validateCreateDeviceParams } from '../validators';
+
+const addDevice: Handler = (
+  req: Request,
+  res: Response,
+): void => {
+  const validation = validateCreateDeviceParams(req.body);
+  if (validation.error) {
+    res.status(400).json({ error: validation.error });
+  } else {
+    deviceService.addDevice(req.body)
+      .then((data) => res.json(data))
```

Comment:
...which means that this will no longer have any data; instead, we can return the original request content:

```suggestion
      .then(() => res.json(req.body))
```